cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -163,14 +171,40 @@ def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBu
             ],
         )
 
+        ami_distribution_configuration = {
+            "Name": self._set_ami_name(),
+            "Description": self.imagebuild.image.description,
+            "AmiTags": ami_tags,
+        }
+
+        imagebuilder.CfnDistributionConfiguration(
+            self,
+            id="ParallelClusterDistributionConfiguration",
+            name="-".join(["ParallelCluster", utils.get_installed_version().replace(".", "-"), resources_prefix]),
+            tags=build_tags,
+            distributions=[
+                imagebuilder.CfnDistributionConfiguration.DistributionProperty(
+                    ami_distribution_configuration=ami_distribution_configuration,
+                    region=core.Fn.sub("${AWS::Region}"),
+                )
+            ],
+        )
+
         # Image
         imagebuilder.CfnImage(
             self,
             id="PClusterImage",
+            tags=build_tags,
             image_recipe_arn=core.Fn.ref("PClusterImageRecipe"),
             infrastructure_configuration_arn=core.Fn.ref("PClusterImageInfrastructureConfiguration"),
+            distribution_configuration_arn=core.Fn.ref("ParallelClusterDistributionConfiguration"),
         )
 
+    def _set_ami_name(self):
+        if "{{ imagebuilder:buildDate }}" not in self.imagebuild.image.name:
+            return self.imagebuild.image.name + " {{ imagebuilder:buildDate }}"
```

Comment:
should we give aws-parallelcluster prefix for custom ami? like `aws-parallelcluster-{{name}}-202012152217`